### PR TITLE
Add a check for configuration added to project.pbxproj

### PR DESCRIPTION
**Pull request by ayoy**
Task URL: https://app.asana.com/0/1203301625297703/1203780951670808/f
iOS PR: https://github.com/duckduckgo/iOS/pull/1464
macOS PR: https://github.com/more-duckduckgo-org/macos-browser/pull/941

Since this makes sense only for macOS Browser for now, the check verifies repo name, and only performs any action for "macos-browser”.

### CODE REVIEW

File: tests/xcodeprojConfiguration.allPRs.test.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    it("does not fail with no diff in project file", async () => {
+        dm.danger.git.diffForFile = async (_filename) => {}
+
+        await xcodeprojConfiguration()
+
+        expect(dm.fail).not.toHaveBeenCalled()
+    })
+
+    it("does not fail with no additions", async () => {
+
```
Comment (loremattei) [inline tests/xcodeprojConfiguration.allPRs.test.ts]: Is this empty line intentional?

File: org/allPRs.ts [hunk 2]
```diff
@@ -16,8 +16,29 @@ export const internalLink = async () => {
     }
 }
 
+export const xcodeprojConfiguration = async () => {
+    if (danger.github.thisPR.repo == "macos-browser") {
```
Comment (loremattei) [inline org/allPRs.ts:20]: If, in the future, we'll have more rules that apply only to some repos, we may want to move the check to the default function to make it easier to figure out what rules are called for a certain repo, but, for now, I think this works better.

**Review by loremattei — APPROVED** [review]
@ayoy this looks great to me. 
Thanks for adding this check. 
🚀

File: tests/xcodeprojConfiguration.allPRs.test.ts (hunk 1 above)
Comment (ayoy) [reply tests/xcodeprojConfiguration.allPRs.test.ts]: Not necessarily, I’ll remove it :)

File: org/allPRs.ts (hunk 2 above)
Comment (ayoy) [reply org/allPRs.ts:20]: Oh yes, that’s a great idea for later 👍

**Pull request merged**